mqtt/mqtt-bridge/src/rpc.rs

```diff
@@ -0,0 +1,504 @@
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+use bson::{doc, Document};
+use bytes::{buf::BufExt, Bytes};
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use tracing::{error, warn};
+
+// Import and use mocks when run tests, real implementation when otherwise
+#[cfg(test)]
+pub use mqtt3::{
+    MockPublishHandle as PublishHandle, MockUpdateSubscriptionHandle as UpdateSubscriptionHandle,
+};
+#[cfg(not(test))]
+pub use mqtt3::{PublishHandle, UpdateSubscriptionHandle};
+
+use mqtt3::{
+    proto::Publication, proto::QoS, proto::SubscribeTo, Event, PublishError,
+    SubscriptionUpdateEvent,
+};
+
+use crate::client::EventHandler;
+
+/// MQTT client event handler to react on RPC commands that EdgeHub sends
+/// to execute.
+///
+/// The main purpose of this handler is to establish a communication channel
+/// between EdgeHub and the upstream bridge.
+/// EdgeHub will use low level commands SUB, UNSUB, PUB. In turn the bridge
+/// sends corresponding MQTT packet to upstream broker and waits for an ack
+/// from the upstream. After ack is received it sends a special publish to
+/// downstream broker.
+pub struct RpcHandler {
+    upstream_subs: UpdateSubscriptionHandle,
+    upstream_pubs: PublishHandle,
```

Comment:
there might be a complication when the client needs to be recreated because of an error it can't recover from. It's not handled yet, but the plan is to recreate the pump. So maybe the rpc handle should send the event to the pump instead of having the public and sub handles.

Reply:
Shouldn't we re-create handler in this case?

Reply:
If the message is passed to the pump then the handle doesn't need to be recreated.

Reply:
Discussed offline with Anca. There are more changes required. Will address them in following PRs